src/common/token/baseEth.ts

```diff
@@ -33,4 +33,46 @@ export class BaseEthToken extends EthereumToken {
       pollingOptions,
     });
   }
+
+  public async pollForTxBeingAvailable({
+    txId,
+  }: {
+    txId: string;
+  }): Promise<void> {
+    this.logger.debug('Polling for transaction to be available on chain', {
+      txId,
+      ...this.pollingOptions,
+    });
+    await new Promise((resolve) =>
+      setTimeout(resolve, this.pollingOptions.initialBackoffMs),
+    );
+
+    let attempts = 0;
```

Comment:
Nit: I think it can be a bit more relaxing on the brain to move away from 0-indexing when conveying attempt counts. e.g. start at 1 and loop over `attempt <= maxAttempts`. Also make logging out the value more natural.